src/Data/V/Linear/Internal.hs

```diff
@@ -125,13 +121,44 @@ instance (1 <= n, n ~ Arity b (a %1 -> f), Elim (n - 1) a b f) => Elim n a b (a
 cons :: forall n a. a %1 -> V (n - 1) a %1 -> V n a
 cons = Unsafe.toLinear2 $ \x (V v) -> V (Vector.cons x v)
 
--- | Creates a 'V' of the specified size by consuming a 'Replicator'.
-fromReplicator :: forall n a. (KnownNat n, Replicator.Elim n a (V n a) (FunN n a (V n a))) => Replicator a %1 -> V n a
-fromReplicator = Replicator.elim @n @a @(V n a) @(FunN n a (V n a)) (make @n @a)
-
--- | Produces a @'V' n a@ from a 'Dupable' value @a@.
-dupV :: forall n a. (KnownNat n, Dupable a, Replicator.Elim n a (V n a) (FunN n a (V n a))) => a %1 -> V n a
-dupV = fromReplicator . dupR
+-- | Builds a n-ary constructor for @'V' n a@ (i.e. a function taking @n@ linear
+-- arguments of type @a@ and returning a @'V' n a@).
+make :: forall n a f. Make n n a f => f
+make = make' @n @n id
+{-# INLINE make #-}
+
+-- | @'Make' m n a f@ asserts that @f@ is a function taking @m@ linear arguments
+-- of type @a@ and then returning a value of type @'V' n a@.
+-- It is solely used to define the type of the 'make' function.
+--
+-- The idea behind 'Make' \/ @make\'@ \/ 'make' is the following:
+--
+-- @f@ takes @m@ values of type @a@, but returns a @'V' n a@ (with @n@ ≥ @m@),
+-- so the @n - m@ missing values must be supplied via the accumulator.
+--
+-- @make\'@ is initially called with @m = n@ via 'make', and as @m@ decreases,
+-- the number of lambda on the left increases and the captured values are put
+-- in the accumulator
+-- (@V[ ... ] \<\>@ represents the "extend" operation for 'V'):
+--
+-- >     make @n
+-- > --> make' @n @n (V[] <>)
+-- > --> λx. make' @(n - 1) @n (V[x] <>)
+-- > --> λx. λy. make' @(n - 2) @n (V[x, y] <>)
+-- > --> ...
+-- > --> λx. λy. ... λz. make' @0 @n (V[x, y, ... z] <>)    -- make' @0 @n f = f V[]
+-- > --> λx. λy. ... λz. V[x, y, ... z]
```

Comment:
I think that this documentation should be on the `make` function, not the `Make` class.

Plus it speaks of `make'`, which isn't exported. Let's avoid this. If you need to document the implementation, do so outside of the docstring, in regular comments.

Remember to give instantiations of the type of `make` with concrete values. Examples are very helpful to disentangle very general types.